opcpa-tpr-config/opcpa-tpr-config.py

```diff
@@ -0,0 +1,164 @@
+import yaml
+
+from functools import partial
+
+from qtpy.QtWidgets import QGridLayout
+
+from ophyd import EpicsSignal
+
+from pydm import Display
+from pydm.widgets import PyDMLabel, PyDMPushButton, PyDMNTTable
+
+
+class App(Display):
+    def __init__(self, parent=None, args=None, macros=None):
+        # print(f'args={args}, macros={macros}')
+
+        # Read in config file
+        # TODO add argparse, make this an arg
+        cfg_file = "opcpa-tpr-config/neh_config.yaml"
+
+        config = {}
+        with open(cfg_file, "r") as f:
+            config = yaml.safe_load(f)
+        self.config = config
+
+        print(f"Loaded config {cfg_file}")
+        # print(config)
+
+        # Call super after handling args/macros but before doing pyqt stuff
+        super().__init__(parent=parent, args=args, macros=macros)
+
+        # Now it is safe to refer to self.ui and access your widget objects
+        # It is too late to do any macros processing
+
+        # Setup main areas of GUI
+        self.setup_main()
+
+        # Setup laser specific portions of GUI
+        for laser in self.config["lasers"]:
+            self.setup_rbvs(laser)
+            self.setup_configs(laser)
+
+    def ui_filename(self):
+        return "opcpa-tpr-config.ui"
+
+    def setup_main(self):
+        """
+        Setup the central widgets for the screen.
+        """
+        title = self.findChild(PyDMLabel, "screen_title")
+        if title is not None:
+            title.setText(self.config["main"]["title"])
```

Comment:
this is perfectly valid, but in pydm apps specifically you an also do the following which may be less wordy:
```
self.ui.screen_title.setText(self.config["main"]["title"])
```
I think you can assume the ui file will be well-formed (e.g. you don't need a fallback if the label is missing) but you may want to make fewer assumptions about the yaml file if you'll let other people edit it.